src/router/index.js

```diff
@@ -25,29 +25,54 @@ const routes = [
   {
     path: '/portfolio',
     name: 'Portfolio',
-    component: () => import(/* webpackChunkName: "Portfolio" */ '../views/Portfolio.vue')
+    component: () => import(/* webpackChunkName: "Portfolio" */ '../views/Portfolio.vue'),
+    beforeEnter (to, from, next) {
+      const token = getCookie('token')
+      if (token) {
+        next()
+      } else {
+        next('/login')
+      }
+    }
   },
   {
     path: '/stocks',
     name: 'Stocks',
     component: () => import(/* webpackChunkName: "Stocks" */ '../views/Stocks.vue'),
     beforeEnter (to, from, next) {
-      if (store.state.auth.idToken) {
+      const token = getCookie('token')
+      if (token) {
         next()
       } else {
-        next('/signup')
+        next('/login')
       }
     }
   },
   {
     path: '/login',
     name: 'Login',
-    component: () => import(/* webpackChunkName: "Login" */ '../views/auth/Login.vue')
+    component: () => import(/* webpackChunkName: "Login" */ '../views/auth/Login.vue'),
+    beforeEnter (to, from, next) {
+      const token = getCookie('token')
+      if (!token) {
+        next()
+      } else {
+        next('/stocks')
+      }
+    }
   },
   {
     path: '/signup',
     name: 'SignUp',
-    component: () => import(/* webpackChunkName: "SignUp" */ '../views/auth/SignUp.vue')
+    component: () => import(/* webpackChunkName: "SignUp" */ '../views/auth/SignUp.vue'),
+    beforeEnter (to, from, next) {
```

Comment:
You need to handle `beforeEnter` only for pages that requires auth. You can have one: `router.beforeRouteEach` function instead of duplication for each route like you have right now. Please use `meta` property on route to check if you need to validate it inside beforeRouteEach like this:

```
{
    path: '/login',
    name: 'Login',
    meta: {
      requiredAuth: true;
    }
```

```
beforeRouteEach(to, from, next) {
  if (to.meta.requireAuth) ... do token check here...
  else next()
}
```